sdk/python/kfp/components/_python_op.py

```diff
@@ -640,7 +639,18 @@ def get_serializer_and_register_definitions(type_name) -> str:
     component_spec.implementation=ContainerImplementation(
         container=ContainerSpec(
             image=base_image,
-            command=package_preinstallation_command + ['python3', '-u', '-c', full_source],
+            command=package_preinstallation_command + [
+                'sh',
+                '-ec',
+                # Writing the program code to a file.
+                # This is needed for Python to show stack traces and for `inspect.getsource` to work (used by PyTorch JIT and this module for example).
+                textwrap.dedent('''\
+                    program_path=$(mktemp)
```

Comment:
Will we have a chance to cleanup the temp file?

Reply:
I do not think we need to do that. The temp file is inside the container and will be deleted with the container when the container finishes.